nautobot_ssot/contrib.py

```diff
@@ -0,0 +1,354 @@
+"""This module includes a base adapter and a base model class for interfacing with Nautobot."""
+# pylint: disable=protected-access
+# Diffsync relies on underscore-prefixed attributes quite heavily, which is why we disable this here.
+
+from collections import defaultdict
+from dataclasses import dataclass
+
+import pydantic
+from diffsync import DiffSyncModel, DiffSync
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from typing_extensions import get_type_hints
+
+
+@dataclass
+class CustomFieldAnnotation:
+    """Map a model field to an arbitrary custom field name.
+
+    For usage with `typing.Annotated`.
+
+    This exists to map model fields to their corresponding custom fields. This solves the problem of Python object
+    attributes not being able to include spaces, while custom field names/labels may.
+
+    Example:
+        Given a boolean custom field "Is Global" on the Provider model:
+
+        ```python
+        class ProviderModel(NautobotModel):
+            _model: Provider
+            _identifiers = ("name",)
+            _attributes = ("is_global",)
+
+            name: str
+            is_global: Annotated[bool, CustomFieldAnnotation(name="Is Global")
+        ```
+
+        This then maps the model field 'is_global' to the custom field 'Is Global'.
+    """
+
+    name: str
+
+
+class BaseAdapter(DiffSync):
+    """
+    Adapter for loading data from Nautobot through the ORM.
+
+    This adapter is able to infer how to load data from Nautobot based on how the models attached to it are defined.
+    """
+
+    @staticmethod
+    def _get_parameter_names(diffsync_model):
+        """Ignore the differences between identifiers and attributes, because at this point they don't matter to us."""
+        return list(diffsync_model._identifiers) + list(diffsync_model._attributes)  # pylint: disable=protected-access
+
+    def _load_models(self, diffsync_model):
```

Comment:
it seems to me that the name is misleading. You aren't loading "models" but "objects" (`self._load_single_object`)

Reply:
I don't see any answer to this point